ui.tests/test-module/specs/tabsontop/tabsontop.runtime.spec.js

```diff
@@ -0,0 +1,90 @@
+    const childBemBlock = 'cmp-adaptiveform-datepicker';
+
+    let formContainer = null;
+
+    beforeEach(() => {
+        cy.previewForm(pagePath).then(p => {
+            formContainer = p;
+        })
+    });
+
+    const checkHTML = (id, state, view, count) => {
+        const visible = state.visible;
+        const passVisibleCheck = `${visible === true ? "" : "not."}be.visible`;
+        cy.get(`#${id}`)
+            .should(passVisibleCheck)
+            .invoke('attr', 'data-cmp-visible')
+            .should('eq', visible.toString());
+        cy.get(`#${id}`)
+            .invoke('attr', 'data-cmp-enabled')
+            .should('eq', state.enabled.toString());
+        expect(state.items.length, "model has children equal to count").to.equal(count);
+        if (count == 0) {
+            cy.get(`.${childBemBlock}`).should('not.exist');
+        } else {
+            cy.get(`.${childBemBlock}`).should('have.length', count);
+        }
+
+        expect(view.children.length, "tab has children equal to count").to.equal(count);
+        return cy.get(`#${id}`);
+    };
+
+
+    it(" should get model and view initialized properly and parent child relationship is set ", () => {
+        expect(formContainer, "formcontainer is initialized").to.not.be.null;
+        Object.entries(formContainer._fields).forEach(([id, field]) => {
+            expect(field.getId()).to.equal(id);
+            expect(formContainer._model.getElement(id), `model and view are in sync`).to.equal(field.getModel());
+        });
+
+        const tabId = formContainer._model.items[0].id;
+        const datepickerId = formContainer._model.items[0].items[0].id;
+        const tabView = formContainer._fields[tabId];
+        const datepickerView = formContainer._fields[datepickerId];
+        expect(tabView, "tab view is created").to.not.be.null;
+        expect(datepickerView, "tabs child view is created").to.not.be.null;
+        expect(tabView.children[0].id, "tab has reference to child view").to.equal(datepickerId);
+        expect(datepickerView.parentView.id, "date picker has reference to parent panel view").to.equal(tabId);
+    })
+
+    it(" model's changes are reflected in the html ", () => {
+        const tabId = formContainer._model.items[0].id;
+        const model = formContainer._model.getElement(tabId);
+        const tabView = formContainer._fields[tabId];
+        const count = 2;
+        checkHTML(model.id, model.getState(), tabView, count).then(() => {
+            model.visible = false;
+            return checkHTML(model.id, model.getState(), tabView, count);
+        }).then(() => {
+            model.enable = false;
+            return checkHTML(model.id, model.getState(), tabView, count);
+        });
+    });
+
+    it("switch tab in runtime", () => {
+        const tab2 = cy.get('ol li').contains('Date Input 2');
+        tab2.click();
+        tab2.should('have.class','cmp-tabs__tab--active');
```

Comment:
pls check for the tabindex and aria selected attributes.
the tab that is hidden should also be check for the corresponding attributes.

Reply:
done